gulpfile.js

```diff
@@ -0,0 +1,174 @@
+    },
+    html: {
+        index: 'src/dev/index.html'
+    }
+}
+
+// Concat and minification for final DOM files
+ gulp.task('concat-uglify-scripts-js', () =>
+    gulp.src(files.js.scripts)
+  	  .pipe(babel({presets: ['es2015']}))
+  		.pipe(order([
+  			'robot.js',
+  			'drone.js',
+  			'bipedal.js',
+  			'atv.js',
+  			'weapons.js',
+  			'setup.js',
+  			'domevents.j',
+  			'**/*.js'
+  		]))
+  		.on('error',errorLog)
+      .pipe(gulpConcat('temp.js'))
+      .pipe(uglify())
+      .on('error', errorLog)
+      .pipe(rename('script.min.js'))
+      .pipe(gulp.dest('src/build/js'))
+ );
+
+ gulp.task('concat-uglify-vender-js', () =>
+    gulp.src(files.js.vender)
+      .pipe(gulpConcat('tempVender.js'))
+      .pipe(uglify())
+      .on('error', errorLog)
+      .pipe(rename('vender.min.js'))
+      .pipe(gulp.dest('src/build/js'))
+);
+
+gulp.task('concat-uglify-styles-css', () =>
+    gulp.src(files.css.styles)
+      .pipe(gulpConcat('temp.css'))
+      .pipe(cleanCSS({'keepSpecialComments' : 0}))
+      .on('error', errorLog)
+      .pipe(rename('style.min.css'))
+      .pipe(gulp.dest('src/build/css'))
+);
+
+gulp.task('concat-uglify-vender-css', () =>
+    gulp.src(files.css.vender)
+      .pipe(gulpConcat('tempVender.css'))
+      .pipe(cleanCSS({'keepSpecialComments' : 0}))
+      .on('error',errorLog)
+      .pipe(rename('vender.min.css'))
+      .pipe(gulp.dest('src/build/css'))
+);
+
+gulp.task('copy-image-min', () =>
+    gulp.src('src/dev/images/*')
+      .pipe(imagemin())
+      .pipe(gulp.dest('src/build/images'))
+);
+
+gulp.task('copy-html', () =>
+  	gulp.src(files.html.index)
+  		.pipe(gulp.dest('src/build'))
+);
+
+gulp.task('insert-css-js-links-copy-html', ['run-js-files', 'run-css-files','copy-image-min', 'copy-html'], ()=> {
+		var jsMinVender = gulp.src('src/build/js/vender.min.js', {read: false});
+		var jsMinUser = gulp.src('src/build/js/script.min.js', {read: false});
+		var cssMinVender = gulp.src('src/build/css/vender.min.css', {read: false});
+		var cssMinUser = gulp.src('src/build/css/style.min.css', {read: false});
+
+		gulp.src('src/build/index.html')
+			.pipe(inject(series(cssMinVender, cssMinUser), {ignorePath: 'src/build/', addRootSlash: false}))
+			.pipe(inject(series(jsMinVender, jsMinUser), {ignorePath: 'src/build/', addRootSlash: false}))
+			.on('error', errorLog)
+			.pipe(gulp.dest('src/build'));
+});
+
+///////////////////////////////////////////////////////////
+
+// Lint task for JS source files
+ gulp.task('lint-js', () =>
+    gulp.src(files.js.scripts)
+      .pipe(jshint({'esversion': 6}))
+      .pipe(jshint.reporter('default'), {verbose: true})
+ );
+// Lint task for JS source files
+
+// Babel task for JS source files
+ gulp.task('babel-js',() =>
+    gulp.src(files.js.scripts)
+      .pipe(babel({presets: ['es2015']}))
+      .pipe(gulp.dest('src/dev/babel'))
+ );
+ // Babel task for JS source files
+
```

Comment:
Any reason you're using Babel? At first glace I don't see any ES6 code that wouldn't work in the Browser